nqp/core/scheduler.py

```diff
@@ -233,7 +233,14 @@ def call_scheduled_functions(self, dt: float):
             item.func(now - item.last_ts)
 
             if item.repeat:
-                item.next_ts = item.last_ts + item.interval
+                """
+                If the first execution was at time == 0.001 and an interval of 1,
+                using item.next_ts = item.last_ts + item.interval the next interval
+                would be at 1 and the last would be 0.001, and after that 
+                next interval would be 1.001 instead of 2, so I think this change 
+                makes sense? Let me know if I'm wrong here
+                """
+                item.next_ts += item.interval
```

Comment:
the scheduler should be resistant against drift.  the reason that `item.last_ts + item.interval` is used is because we want to make sure that interval scheduled items are repeated in a consistent manner.  if we do `item.next_ts += item.interval`, then any error introduced in the last frame will accumulate over time.

Reply:
But isn't going from 0.001 to 1 to 1.001 to 2 accumulating errors?

Reply:
consider the following:
```
1. expected execution |----X----X----X----X----X--
2. missed frames      |---------XX----------------
3. "add last_ts"      |----X------X--X----X----X--
4. "add interval"     |----X------X----X----X----X

```
we want 1, but the game lags and we miss a frame.  in 2, we miss 2 frames at X, which is also the same frame we want a execution.  we missed it, so we run it late.  if we run late and add the interva (4)l, we have now offset the timing by one frame forever.  the original timing is now offset.  but if we use "last_ts" (3), and add the interval, we effectively subract the missed frame from the interval and we stay in the original timing, on line 1